packages/grafana-i18n/src/i18n.tsx

```diff
@@ -140,29 +167,38 @@ export function addResourceBundle(language: string, namespace: string, resource:
   getI18nInstance().addResourceBundle(language, namespace, resource, undefined, true);
 }
 
+function logError(entity: string) {
+  const message = `${entity} was called before i18n was initialized. This is probably caused by calling ${entity} outside of Grafana context or by calling ${entity} in the root module scope, instead of lazily on render. Initializing default instances.`;
+  console.error(message);
```

Comment:
By logging an error, this would be picked up by our frontend logging and we could create dashboard to find these occasions out in the wild. Should also automatically break any e2e tests where this happens.